Sources/xcodeproj/Objects/Project/PBXObjectParser.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+final class PBXObjectParser {
+    
+    let reference: String
+    let userInfo: [CodingUserInfoKey: Any]
+    
+    init(reference: String, userInfo: [CodingUserInfoKey: Any]) {
+        self.reference = reference
+        self.userInfo = userInfo
+    }
+    
+    // swiftlint:disable function_body_length
+    public func parse(dictionary: [String: Any]) throws -> PBXObject {
+        let decoder = XcodeprojJSONDecoder()
```

Comment:
Looking at the code I think there's a bug here *(not introduced by the PR)*. The decoder, has a context object with a property `objectReferenceRepository` which is used by the objects when they are being initialised to create the references with other objects. When we create a decoder for each object being parsed, we are passing an empty reference repository.

We can fix it by moving the decoder to an instance variable of `PBXObjectParser`, and change the logic that iterates through the keys and values of the pbxproj to use the same parser instead of creating an instance in every step of the iteration.